src/components/SchemaTable/index.jsx

```diff
@@ -50,12 +50,13 @@ const useStyles = makeStyles(theme => ({
    * in the left panel of the schema table component.
    */
   code: {
-    backgroundColor: theme.palette.grey[300],
+    backgroundColor: theme.palette.text.primary,
+    color: theme.palette.getContrastText(theme.palette.text.primary),
     padding: `0 ${theme.spacing(0.5)}px`,
   },
   /** Comments within the left panel (used for combination types) */
   comment: {
-    color: theme.palette.text.hint,
+    color: theme.palette.text.primary,
```

Comment:
What was the reason for changing `hint` to `primary`?

Reply:
Using `hint`, the comments are not visible against a dark background. So the change result would be

| before | after |
|:---:|:---:|
| <img src="https://user-images.githubusercontent.com/29671309/74515258-9345ec80-4f51-11ea-9891-aa46a1b5ad28.png" height="200px" /> | <img src="https://user-images.githubusercontent.com/29671309/74515151-5a0d7c80-4f51-11ea-8a5d-4ab63becf50a.png" height="200px" /> |

Alternatively, we could leave comment color configuration to the user, so that it is the user's responsibility to configure the theme colors so that the comments are visible. Would this be better than to use `primary` ?

Reply:
If we just use the default dark theme with

```jsx
   const colorfulTheme = createMuiTheme({
    palette: {
      type: 'dark'
    },
  });
```

then the comment will be readable:

![Screen Shot 2020-02-14 at 3 15 10 PM](https://user-images.githubusercontent.com/3766511/74564677-99ca6900-4f3d-11ea-9a97-85d9ca844686.png)

If the user decides to change the default dark theme to something else like

```jsx
  const colorfulTheme = createMuiTheme({
    palette: {
      text: {
        primary: '#ffc107',
      },
    },
  });
```

then they would also have to change other theme values to make it sit nicely. Here's a quick example that's incomplete. Let me know what you think.

```diff
diff --git a/src/components/Header/index.jsx b/src/components/Header/index.jsx
index e4b7b72..408eeb6 100644
--- a/src/components/Header/index.jsx
+++ b/src/components/Header/index.jsx
@@ -7,7 +7,7 @@ import { schema } from '../../utils/prop-types';
 
 const useStyles = makeStyles(theme => ({
   container: {
-    backgroundColor: theme.palette.getContrastText(theme.palette.text.primary),
+    backgroundColor: theme.palette.background.paper,
     color: theme.palette.text.primary,
     padding: `${theme.spacing(1.5)}px ${theme.spacing(1)}px`,
     marginBottom: theme.spacing(1),
diff --git a/src/components/SchemaTable/index.jsx b/src/components/SchemaTable/index.jsx
index ec6ab9d..4b18ae9 100644
--- a/src/components/SchemaTable/index.jsx
+++ b/src/components/SchemaTable/index.jsx
@@ -16,13 +16,13 @@ const useStyles = makeStyles(theme => ({
   },
   /** The left panel of the Schema Table */
   leftPanel: {
-    backgroundColor: theme.palette.getContrastText(theme.palette.text.primary),
+    backgroundColor: theme.palette.background.paper,
     borderRight: `${theme.spacing(1)}px solid ${theme.palette.divider}`,
     overflowX: 'auto',
   },
   /** The right panel of the Schema Table */
   rightPanel: {
-    backgroundColor: theme.palette.getContrastText(theme.palette.text.primary),
+    backgroundColor: theme.palette.background.paper,
     overflowX: 'auto',
   },
   /** Rows for the left and right panels */
@@ -53,7 +53,7 @@ const useStyles = makeStyles(theme => ({
    * in the left panel of the schema table component.
    */
   code: {
-    backgroundColor: theme.palette.grey[300],
+    backgroundColor: theme.palette.getContrastText(theme.palette.text.primary),
     padding: `0 ${theme.spacing(0.5)}px`,
   },
   /** Comments within the left panel (used for combination types) */
@@ -73,7 +73,7 @@ const useStyles = makeStyles(theme => ({
    */
   chip: {
     color: theme.palette.text.primary,
-    borderColor: theme.palette.text.primary
+    borderColor: theme.palette.text.secondary,
   },
 }));
 
diff --git a/src/components/SchemaViewer/index.stories.js b/src/components/SchemaViewer/index.stories.js
index fce1803..9e82e3e 100644
--- a/src/components/SchemaViewer/index.stories.js
+++ b/src/components/SchemaViewer/index.stories.js
@@ -138,9 +138,15 @@ export const customThemes = () => {
   });
   const colorfulTheme = createMuiTheme({
     palette: {
+      background: {
+        paper: '#000',
+      },
       text: {
         primary: '#ffc107',
+        secondary: '#ffc53d',
+        hint: '#ddd',
       },
+      divider: '#4f4f4f',
     },
   });
 
diff --git a/src/components/SourceView/index.jsx b/src/components/SourceView/index.jsx
index 7ce2997..a4885f8 100644
--- a/src/components/SourceView/index.jsx
+++ b/src/components/SourceView/index.jsx
@@ -4,7 +4,7 @@ import { schema } from '../../utils/prop-types';
 
 const useStyles = makeStyles(theme => ({
   view: {
-    backgroundColor: theme.palette.background.default,
+    backgroundColor: theme.palette.background.paper,
     color: theme.palette.text.primary,
     overflowX: 'auto',
   },
```

Dark theme:
![Screen Shot 2020-02-14 at 3 55 51 PM](https://user-images.githubusercontent.com/3766511/74566903-8a99ea00-4f42-11ea-83a3-53ac0f0d479c.png)

Colorful theme:
![Screen Shot 2020-02-14 at 3 56 06 PM](https://user-images.githubusercontent.com/3766511/74566905-8b328080-4f42-11ea-9650-9fb2e3e42542.png)

Reply:
Sounds great! 👍 Applied changes!